testing/integration/agent_long_running_leak_test.go

```diff
@@ -0,0 +1,296 @@
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	testRuntime := os.Getenv("LONG_TEST_RUNTIME")
+	if testRuntime == "" {
+		testRuntime = "15m"
+	}
+
+	regex := regexp.MustCompile(`[\d]+`)
+	status, err := runner.agentFixture.ExecStatus(ctx)
+	require.NoError(runner.T(), err)
+
+	// because we need to separately fetch the PIDs, wait until everything is healthy before we look for running beats
+	require.Eventually(runner.T(), func() bool {
+		allHealthy := true
+		status, err := runner.agentFixture.ExecStatus(ctx)
+
+		cefMatch := "logfile-apache"
+		foundCef := false
+		systemMatch := "metrics-default"
+		foundSystem := false
+
+		require.NoError(runner.T(), err)
+		for _, comp := range status.Components {
+			// make sure the components include the expected integrations
+			for _, v := range comp.Units {
+				runner.T().Logf("unit ID: %s", v.UnitID)
+				// the full unit ID will be something like "log-default-logfile-cef-3f0764f0-4ade-4f46-9ead-f2f0f7865676"
+				if !foundCef && strings.Contains(v.UnitID, cefMatch) {
+					foundCef = true
+				}
+				if !foundSystem && strings.Contains(v.UnitID, systemMatch) {
+					foundSystem = true
+				}
+			}
+			runner.T().Logf("component state: %s", comp.Message)
+			if comp.State != int(cproto.State_HEALTHY) {
+				allHealthy = false
+			}
+		}
+		return allHealthy && foundCef && foundSystem
+	}, time.Minute*3, time.Second*20, "install never became healthy")
+
+	handles := []processWatcher{}
+
+	// track running beats
+	// the `last 30s` metrics tend to report gauges, which we can't use for calculating a derivative.
+	// so separately fetch the PIDs
+	status, err = runner.agentFixture.ExecStatus(ctx)
+	require.NoError(runner.T(), err)
+	for _, comp := range status.Components {
+		pidStr := regex.FindString(comp.Message)
+		pid, err := strconv.ParseInt(pidStr, 10, 64)
+		require.NoError(runner.T(), err)
+
+		handle, err := sysinfo.Process(int(pid))
+		require.NoError(runner.T(), err)
+		handlesReg := tools.NewSlope(fmt.Sprintf("%s handle usage", comp.Name))
+
+		runner.T().Logf("created handle watcher for %s (%d)", comp.Name, pid)
+		handles = append(handles, processWatcher{handle: handle, pid: int(pid), name: comp.Name, regHandles: handlesReg})
+	}
+
+	testDuration, err := time.ParseDuration(testRuntime)
+	require.NoError(runner.T(), err)
+
+	timer := time.NewTimer(testDuration)
+	defer timer.Stop()
+
+	// time to perform a health check
+	ticker := time.NewTicker(time.Second * 10)
+	defer ticker.Stop()
+
+	done := false
+	start := time.Now()
+	for !done {
+		select {
+		case <-timer.C:
+			done = true
+		case <-ticker.C:
+			err := runner.agentFixture.IsHealthy(ctx)
+			require.NoError(runner.T(), err)
+			// for each running process, collect memory and handles
+			for _, handle := range handles {
+
+				ohc, ok := handle.handle.(types.OpenHandleCounter)
+				if ok {
+					handleCount, err := ohc.OpenHandleCount()
+					require.NoError(runner.T(), err)
```

Comment:
Do we fail the test and stop immediately if we fail to fetch one data point ?
If we use `assert` instead of `require` we could have more insight on how often this happens and keep running until the end to see if there's a trend or other issues
Same for the require in line 254

Reply:
So, that call isn't really expected to fail under a normal workload; a failure most likely means that the process doesn't exist or is no longer runner, which probably is an  immediate failure.